### PR TITLE
Zio raft tests

### PR DESCRIPTION
**Pull request by kaplan-shaked**

### CODE REVIEW

File: calculator/src/main/scala/zio/calculator/App.scala [hunk 1]
```diff
@@ -72,6 +72,5 @@ object CalculatorApp extends zio.App:
     raft.use(raft => {
       for 
         _ <- new HttpServer(raft).run.fork.when(memberId == MemberId("peer1"))
-        _ <- raft.run
```
Comment (somdoron) [inline calculator/src/main/scala/zio/calculator/App.scala:75]: if you remove this, the above should not be forked

**Review by somdoron — APPROVED** [review]

File: calculator/src/main/scala/zio/calculator/App.scala (hunk 1 above)
Comment (kaplan-shaked) [reply calculator/src/main/scala/zio/calculator/App.scala:75]: I add inside the makeManaged with forkManaged.

File: raft/src/main/scala/zio/raft/Raft.scala [hunk 2]
```diff
@@ -558,3 +563,5 @@ object Raft:
       refStateMachine,
       pendingCommands
     )
+      _ <- raft.run.forkManaged
```
Comment (kaplan-shaked) [inline raft/src/main/scala/zio/raft/Raft.scala:566]: @somdoron run it inside the makeManaged.

Comment (somdoron) [reply raft/src/main/scala/zio/raft/Raft.scala:566]: understood, but now the `use` in the calculator app main will exit immediately.

**Pull request merged**